tests/scripts/run-xunit-perf.cmd

```diff
@@ -2,193 +2,294 @@
-                                    --config OS "Windows_NT" ^
-                                    --arch "%TEST_ARCHITECTURE%" ^
-                                    --machinepool "PerfSnake"
-  py "%BENCHVIEW_PATH%\upload.py" submission.json --container coreclr
-)
-
-goto :EOF
-
-:DOIT
-set BENCHNAME=%~n1
-set BENCHDIR=%~p1
-set PERFOUT=perf-%BENCHNAME%
-set XMLOUT=%PERFOUT%-summary.xml
-
-echo --- Running %BENCHNAME%
-
-@rem copy benchmark and any input files
-
-xcopy /s %1 . >> %RUNLOG%
-xcopy /s %BENCHDIR%*.txt . >> %RUNLOG%
-
-set CORE_ROOT=%CORECLR_REPO%\sandbox
-
-@rem setup additional environment variables
-if DEFINED TEST_ENV (
-    if EXIST !TEST_ENV! (
-        call %TEST_ENV%
+@if defined _echo echo on
+
+setlocal
+  set ERRORLEVEL=
+  set CORECLR_REPO=%CD%
+  set TEST_FILE_EXT=exe
+  set TEST_ARCH=x64
+  set TEST_ARCHITECTURE=x64
+  set TEST_CONFIG=Release
+  set IS_SCENARIO_TEST=
+  set USAGE_DISPLAYED=
+
+  call :parse_command_line_arguments %*
+  if defined USAGE_DISPLAYED exit /b %ERRORLEVEL%
+
+  call :set_test_architecture || exit /b 1
+  call :verify_core_overlay   || exit /b 1
+  call :set_perf_run_log      || exit /b 1
+  call :setup_sandbox         || exit /b 1
+
+  call :run_cmd "%CORECLR_REPO%\Tools\dotnetcli\dotnet.exe" restore "%CORECLR_REPO%\tests\src\Common\PerfHarness\project.json"                                        || exit /b 1
+  call :run_cmd "%CORECLR_REPO%\Tools\dotnetcli\dotnet.exe" publish "%CORECLR_REPO%\tests\src\Common\PerfHarness\project.json" -c Release -o "%CORECLR_REPO%\sandbox" || exit /b 1
+
+  rem TODO: Remove the version of the package to copy. e.g.) if multiple version exist, then error out?
+  call :run_cmd xcopy /sy "%CORECLR_REPO%\packages\Microsoft.Diagnostics.Tracing.TraceEvent\1.0.3-alpha-experimental\lib\native"\* . >> %RUNLOG%  || exit /b 1
+  call :run_cmd xcopy /sy "%CORECLR_REPO%\bin\tests\Windows_NT.%TEST_ARCH%.%TEST_CONFIG%\Tests\Core_Root"\* . >> %RUNLOG%                         || exit /b 1
+
+  rem find and stage the tests
+  for /R %CORECLR_PERF% %%T in (*.%TEST_FILE_EXT%) do (
+    call :run_benchmark %%T
+  )
+
+  rem optionally upload results to benchview
+  if not [%BENCHVIEW_PATH%] == [] (
+    call :upload_to_benchview
+  )
+
+  exit /b %ERRORLEVEL%
+
+:run_benchmark
+rem ****************************************************************************
+rem   Executes the xUnit Performance benchmarks
+rem ****************************************************************************
+setlocal
+  set BENCHNAME=%~n1
+  set BENCHDIR=%~p1
+  set PERFOUT=perf-%BENCHNAME%
+  set XMLOUT=%PERFOUT%.xml
+
+  rem copy benchmark and any input files
+  call :run_cmd xcopy /s %~1 . >> %RUNLOG%  || exit /b 1
+  if exist "%BENCHDIR%*.txt" (
+    call :run_cmd xcopy /s %BENCHDIR%*.txt . >> %RUNLOG%  || exit /b 1
+  )
+
+  set CORE_ROOT=%CORECLR_REPO%\sandbox
+
+  rem setup additional environment variables
+  if DEFINED TEST_ENV (
+    if EXIST "%TEST_ENV%" (
```

Comment:
This should now work as it is not inside a code block that would evaluate before it is set.